src/main/webapp/app/components/tabs/ReviewHistoryTab.tsx

```diff
@@ -0,0 +1,89 @@
+import { GET_ALL_DRUGS_PAGE_SIZE } from 'app/config/constants/constants';
+import { FB_COLLECTION } from 'app/config/constants/firebase';
+import { HistoryCollection } from 'app/shared/model/firebase/firebase.model';
+import { downloadFile } from 'app/shared/util/file-utils';
+import { getAllGeneHistoryForDateRange, getHistoryEntryStrings } from 'app/shared/util/firebase/firebase-history-utils';
+import { componentInject } from 'app/shared/util/typed-inject';
+import { IRootStore } from 'app/stores';
+import { onValue, ref } from 'firebase/database';
+import React, { useEffect, useState } from 'react';
+import { Button, Input, Label, Row, Spinner } from 'reactstrap';
+
+function ReviewHistoryTab({ firebaseDb, drugList, getDrugs }: StoreProps) {
+  const [historyCollection, setHistoryCollection] = useState<HistoryCollection>(null);
+  const [startDate, setStartDate] = useState<string>('');
+  const [endDate, setEndDate] = useState<string>('');
+
+  useEffect(() => {
+    const unsubscribe = onValue(ref(firebaseDb, FB_COLLECTION.HISTORY), snapshot => {
+      setHistoryCollection(snapshot.val());
+    });
+
+    return () => unsubscribe?.();
+  }, []);
+
+  useEffect(() => {
+    getDrugs({ page: 0, size: GET_ALL_DRUGS_PAGE_SIZE, sort: 'id,asc' });
+  }, []);
+
+  function handleDownload() {
+    const start = startDate ? new Date(startDate) : null;
+    const end = endDate ? new Date(endDate) : null;
+
+    const history = getAllGeneHistoryForDateRange(historyCollection, drugList, start, end);
```

Comment:
Can this fail to get the history?

Reply:
I show a loading indicator while fetching the data from firebase and disable the download button. Since that is inside the click handler, a the history must be present for that function to be called.

It is possible for fetching the firebase data to fail, which will result in the user seeing a loading indicator. However, since it's a listener, I assume it will retry fetching. Throughout the project, we haven't really handled for individual listeners failing. I think in that case we have bigger problems (firebase failing), so I think it might be okay to ignore that possibility here.